### DISCUSSION TITLE
Smagorinsky model: grad_u or strain rate tensor?

### DISCUSSION THREAD

**Discussion by Eilloo** [idaholab/moose#22733] [Q&A Modules: Navier-Stokes] [answered]
Hello all,

I've been looking into the existing INSADSmagorinskyEddyViscosity kernel, with a view to try implementing Van Driest damping in the future.

I'm no expert in turbulence modelling and LES, and am having trouble understanding the term which is added to the momentum equation for this kernel. In the documentation, I believe this Reynolds stress term is:

$\nabla\cdot\left ( -\nu_{SGS}\rho\nabla\overrightarrow{u} \right )$

From my own reading, I would expect the sub-grid scale viscosity to be multiplied by the strain rate tensor, rather than grad_u. Generally speaking, I was under the impression that shear stress in a newtonian fluid is proportional to the strain rate, which in 3D is not necessarily the same as the velocity gradient.

I have also encountered a derivation for Reynolds stresses which leads to the following:

$\tau_{SGS} = 2\rho\nu_{SGS}(S_{ij}-\frac{1}{3}\frac{\partial u_{k}}{\partial x_{k}}\delta_{ij}) - \frac{2}{3}\rho k_{SGS}\delta_{ij}$

Where $\delta_{ij}$ is Kronecker delta. I gather the additional terms on the RHS, besides the strain rate tensor, are correcting for some error which would otherwise be associated with this approach. How $\nu_{SGS}$ is determined differs depending on which turbulence model is being implemented. For the Smagorinsky model, we would define the sub-grid scale viscosity as in the documentation, and substitute into this equation to find the Reynolds stress term.

I'm hoping someone can help reconcile these different approaches: Viscosity_SGS * grad_u, Viscosity_SGS * StrainRateTensor, and the last equation with the error correction terms, and help me understand why the form given in the documentation is used.

Thanks!

**Comment by Eilloo** [comment, marked as answer]
@gridley, I wonder if you can help with this? Having thought about it some more, I'm thinking that all these additional terms are related to changes in volume and therefore disappear for the incompressible case. I would be grateful if you can confirm whether my understanding is correct or if there's something else going on which I have missed!

**Reply by gridley** [reply]
Hey, so, this was just a little experiment for a class project that I included in a PR as a starting point for others. It is completely unvalidated, because I found MOOSE to be too slow for running 3D LES just given the lack of ability to do efficient preconditioning for this type of system. I had based the formulation off of what was given in this paper:

Bouffanais, Roland, Michel O. Deville, and Emmanuel Leriche. “Large-Eddy Simulation of the Flow in a Lid-Driven Cubical Cavity.” Physics of Fluids 19, no. 5 (May 1, 2007): 055108. https://doi.org/10.1063/1.2723153.

And yeah, my attention was only being paid to incompressible flows at the time. In that paper, you can see that the eddy viscosity is indeed just summed with the normal viscosity. Also, you have to mind that what's implemented in the kernel is the weak form, so gradient multiplied by the test function gradient corresponds to a laplacian in the strong form PDE.

**Reply by GiudGiud** [reply]
What is implemented in the INSADSmagorinskyEddyViscosity is equation 4 of the paper Gavin linked to.
The normal viscosity can be added with an INSADMomentumViscous kernel.

The Reynolds stresses are simplified in the incompressible Smagorinsky model

**Reply by gridley** [reply]
Thanks @GiudGiud !

**Reply by Eilloo** [reply]
Ok, so to make sure I'm understanding this correctly:
Whilst the Smagorinsky model uses the concept of turbulent viscosity, it does not use the equation for $\tau_{SGS}$ in my post above. 
Rather, when equation 4 in the cited paper for $\nu_{SGS}$ is derived, this is done in such a way that $\nu_{SGS}$ can simply be added to the usual viscosity in the Navier-Stokes momentum equations, and hence model the effect of sub-grid scale turbulent stresses.

@gridley, regarding MOOSE running 3D LES slowly - is this specifically for LES simulations, or is this a general comment about the Navier-Stokes module? Possibly unrelated if the issues are to do with preconditioning rather than mesh resolution requirements, but I've been mulling over how to an appropriate grid size, given the absence of a k-epsilon or k-omega model to estimate the integral length scale. 

Thanks both for your help in clarifying what's currently implemented - hopefully this will enable me to properly include some damping functions for the existing kernel.

**Reply by GiudGiud** [reply]
@Eilloo are you interested in using finite volume NS instead of FE?
We have a K-epsilon turbulence model there. It's not merged yet, it's in this PR
https://github.com/idaholab/moose/pull/22228

Scalable performance on NS simulations can only be achieved with field splits so far. We are in the process of deploying:
- a fix for vector variables field splits (was 3D LES using vector variables?) https://github.com/idaholab/moose/pull/22643
- some working field splits for NSFV https://github.com/idaholab/moose/pull/22405

Beyond field split preconditioning of the fully coupled problem, we are also deploying segregated solution schemes (as opposed to fully coupled) in a single application https://github.com/idaholab/moose/pull/22699

**Reply by Eilloo** [reply]
I'm trying to stay in FE as the plan is to look at fluid-structure interactions, and therefore make use of the FSI module in MOOSE. From my scouring around the documentation and forums, I gather coupling between finite element and finite volume, as well as mesh morphing in finite volume, are both difficult problems to get around?
This is one of my motivations to try developing the existing turbulence model in FE to get some more accurate results at higher Reynolds numbers.

That said, it's great to hear that k-epsilon is on its way in FVNS, as I am also exploring some finite volume bits in the background.

Regarding performance scaling, that's good to know. Most of my simulations thus far have been relatively simple 2D cases, but I will experiment with field splits as these become more complex and start taking longer to run.
